### PR TITLE
AWS Nomad ASG

### PR DESCRIPTION
**Pull request by ryanwohara**
:gear: **Issue**

https://circleci.atlassian.net/browse/SERVER-1308

We don't yet have an ASG for Nomad within AWS.

:white_check_mark: **Fix**

Added an ASG.

:question: **Tests**

- [ ] Passed _reality check_
- [X] Terraform ran
- [X] ASG scaled as expected

### CODE REVIEW

File: nomad-aws/main.tf [hunk 1]
```diff
@@ -55,22 +55,43 @@ data "cloudinit_config" "nomad_user_data" {
...
+  force_delete         = true
+
+  tag {
+    key                 = "Name"
+    value               = "${var.basename}-nomad-client"
+    propagate_at_launch = "true"
+  }
+
+  tag {
+    key                 = "team"
```
Comment (nsmith5) [inline nomad-aws/main.tf:93]: This leaks details of our company infrastructure into the product in a weird way. Perhaps its better that we personally use a block like [this one](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/resource-tagging#propagating-tags-to-all-resources) to propagate tags to instances? Or maybe use a variable to pass in instance tags?

**Review by nsmith5 — CHANGES_REQUESTED** [review]

File: nomad-aws/main.tf (hunk 1 above)
Comment (ryanwohara) [reply nomad-aws/main.tf:93]: I'd prefer to use a variable. I was mostly just persisting what we had.

Comment (nsmith5) [reply nomad-aws/main.tf:93]: Yup, variable would work well!

**Comment by ryanwohara** [comment]
Updating the README now...

**Review by nsmith5 — APPROVED** [review]

**Pull request merged**